charts/kong/templates/_helpers.tpl

```diff
@@ -552,6 +567,11 @@ The name of the service used for the ingress controller's validation webhook
     containerPort: {{ .Values.ingressController.admissionWebhook.port }}
     protocol: TCP
   {{- end }}
+  {{ if (eq $controllerIs2xPlus "true") -}}
```

Comment:
can we simplify the `(eq $controllerIs2xPlus "true")` to just `$controllerIs2xPlus` since it's a boolean value already?

Reply:
It _should_ be a boolean value :smile: Earlier versions tried exactly that, causing me a great deal of confusion when debug printouts of the helper result clearly showed that it was `false` but the block inside the `if` still rendered.

As best I can tell through black box analysis (since I didn't want to try and run Helm via delve to inspect the guts of what gotemplate is actually doing) is that the result of an `include` is always a string from the includer's perspective, regardless of what type it was within the `include`.

Thus, if it evaluates to `false`, the actual check in the `if` is a string check, and anything other than the empty string is truthy.

There's no function to convert string to boolean, so the simplest solution is to perform an exact string comparison. It's real stupid and makes me cry out in pain for a proper function call system with type guarantees. 50% of template development is "damnit, I really need to build an operator to replace this."